modules/007-registry-packages-proxy/images/registry-packages-proxy/src/cmd/registry-packages-proxy/main.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2024 Flant JSC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"net"
+	"os"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/alecthomas/kingpin.v2"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	"github.com/deckhouse/deckhouse/go_lib/registry-packages-proxy/proxy"
+
+	"registry-packages-proxy/internal/app"
+	"registry-packages-proxy/internal/cache"
+	"registry-packages-proxy/internal/credentials"
+)
+
+func main() {
+	kpApp := kingpin.New("registry packages proxy", "A proxy for registry packages")
+	kpApp.HelpFlag.Short('h')
+
+	app.InitFlags(kpApp)
+
+	kpApp.Action(func(_ *kingpin.ParseContext) error {
+		ctx := context.Background()
```

Comment:
Please catch and handle INTERRUPT and SIGTERM signals please

Reply:
done